ccdproc/combiner.py

```diff
@@ -132,31 +172,47 @@ def __init__(self, ccd_iter, dtype=None):
                 if not (default_unit == ccd.unit):
                     raise TypeError("CCDData objects don't have the same unit.")
 
+        # Set array namespace
+        xp = xp or array_api_compat.array_namespace(ccd_list[0].data)
+        self._xp = xp
+        if dtype is None:
+            dtype = xp.float64
         self.ccd_list = ccd_list
```

Comment:
Having both ccd_list and _data_arr can double the memory footprint. Are both needed?